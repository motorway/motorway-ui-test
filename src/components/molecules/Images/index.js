import React from "react";
import Image from "../../atoms/Image";
import "./style.css";

const Images = ({ images }) => (
  <div className="images_container">
    {images.map(({ id, url, user: { profile_image: image } }) => (
      <Image key={id} url={url} image={image} />
    ))}
  </div>
);

export default Images;
