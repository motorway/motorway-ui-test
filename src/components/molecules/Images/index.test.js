import React from "react";
import { render, screen } from "@testing-library/react";
import Images from ".";

jest.mock("../../atoms/Image", () => {
  const stubText = "fake image";
  return () => <div> {stubText}</div>;
});

describe("Molecules: Images", () => {
  let props;
  let component;
  beforeEach(() => {
    props = {
      images: [
        {
          id: "sampleId0",
          url: "sampleUrl0",
          user: { profile_image: "sampleImage0" },
        },
        {
          id: "sampleId1",
          url: "sampleUrl1",
          user: { profile_image: "sampleImage1" },
        },
      ],
    };
    component = render(<Images {...props} />);
  });
  it("Should match snapshot", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
  it("Should contain required amount of image elements", async () => {
    const images = await screen.findAllByText("fake image");
    const expected = 2;
    expect(images.length).toEqual(expected);
  });
});
