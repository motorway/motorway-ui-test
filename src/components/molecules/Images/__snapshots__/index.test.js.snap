// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules: Images Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="images_container"
  >
    <div>
       fake image
    </div>
    <div>
       fake image
    </div>
  </div>
</DocumentFragment>
`;
